### PR TITLE
Performance Test: Roster Prototype

### PR DESCRIPTION
**Pull request by vyhnalekl**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

This PR adds performance tests to the docs. There is a new prototype called Roster and a perf test created from this prototype.

### CODE REVIEW

File: packages/fluentui/docs/src/prototypes/Roster/index.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+const RosterPrototype: React.FC = () => {
+  return (
+    <PrototypeSection title="Roster">
+      <ComponentPrototype>
+        <Provider theme={teamsDarkTheme}>
+          <Provider theme={themeOverrides}>
+            <Flex column variables={{ isContainer: true }}>
+              <Flex space="between" padding="padding.medium" variables={{ isCallingRosterSectionTitle: true }}>
+                <Header as="h3" content="Participants"></Header>
```
Comment (mnajdova) [inline packages/fluentui/docs/src/prototypes/Roster/index.tsx:35]:
```suggestion
                <Header as="h3" content="Participants" />
```

File: packages/fluentui/docs/src/prototypes/Roster/participant.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      {talking ? (
+        <Animation name="talking">
+          <Avatar variables={{ isTalking: true }} image={image} name={name} status={status} />
+        </Animation>
+      ) : (
+        <Avatar image={image} name={name} status={status} />
+      )}
+
+      <Flex column>
+        {talking ? (
```
Comment (mnajdova) [inline packages/fluentui/docs/src/prototypes/Roster/participant.tsx:26]: You can move this ternary opetarot to just add the `weight` prop, something like:

```tsx
<Text weight={talking && "bold"} content={name} variables={{ isNameText: true }} />
```

File: packages/fluentui/docs/src/prototypes/Roster/participant.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        <Text variables={{ isRoleText: true }} content={role} />
+      </Flex>
+
+      {muted ? (
+        <FlexItem push>
+          <MicOffIcon />
+        </FlexItem>
+      ) : null}
+      {muted ? (
+        <MenuButton
```
Comment (mnajdova) [inline packages/fluentui/docs/src/prototypes/Roster/participant.tsx:40]: You can move the `FlexItem` on line 35 over here, the condition is the same

File: packages/fluentui/docs/src/prototypes/Roster/participant.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                  <MicIcon
+                    onClick={() => {
+                      setMuted(false);
+                      setTalking(isTalking);
+                    }}
+                  />
+                ),
+                content: (
+                  <Text
+                    onClick={() => {
```
Comment (jurokapsiar) [inline packages/fluentui/docs/src/prototypes/Roster/participant.tsx:51]: please move the onClick to the menu item itself, as the text is not focusable, keyboard and screen reader interactions will not work

File: packages/fluentui/docs/src/prototypes/Roster/index.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    {expanded ? <TriangleDownIcon /> : hasSubtree ? <TriangleEndIcon /> : ''}
+    <Text size="small" content={content}></Text>
+  </Component>
+);
+
+const treeItems = [
+  {
+    id: 'participants',
+    title: 'Currently in this meeting',
+    items: participants,
```
Comment (jurokapsiar) [inline packages/fluentui/docs/src/prototypes/Roster/index.tsx:20]: items need to be defined as shorthand collection, not as a collection of components. Please see the Tree compoennt examples, for example in https://fluentsite.z22.web.core.windows.net/components/tree/definition?showCode=true&showRtl=false&showTransparent=false&showVariables=false#types-tree

File: packages/fluentui/docs/src/prototypes/Roster/participantData.tsx [hunk 6]
```diff
@@ -0,0 +1,578 @@
...
+
+const participants = [
+  {
+    id: 1,
+    title: (
+      <Participant
+        name="Irving Kuhic"
+        role="Software Engineer"
+        isMuted
+        image="https://s3.amazonaws.com/uifaces/faces/twitter/ariffsetiawan/128.jpg"
```
Comment (miroslavstastny) [inline packages/fluentui/docs/src/prototypes/Roster/participantData.tsx:13]: For the perf test to be relevant there should be no additional network requests.
Use base64 encoded images, similar to this:
https://github.com/microsoft/fluentui/blob/8247d915e42b98aceeeaaf0061c1ce3773bd17fb/packages/fluentui/docs/src/examples/components/List/Performance/ListCommon.perf.tsx#L4-L34

File: packages/fluentui/docs/src/examples/components/PerformanceTests/Performance/roster.perf.tsx [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import * as React from 'react';
```
Comment (miroslavstastny) [inline packages/fluentui/docs/src/examples/components/PerformanceTests/Performance/roster.perf.tsx:1]: Rename the file to start with a capital 'R' -> `Roster.perf.tsx`

**Comment by ecraig12345** [comment]
@vyhnalekl Is this still needed?

**Comment by miroslavstastny** [comment]
Go to Performance Tests, Show prototype, click Performance Tests again - it crashes:
output

File: packages/fluentui/docs/src/components/Sidebar/Sidebar.tsx [hunk 8]
```diff
@@ -561,6 +566,11 @@ const Sidebar: React.FC<RouteComponentProps & SidebarProps> = props => {
           setActiveItemIds(activeItemIds);
         }}
       />
+      <Flex column>
```
Comment (miroslavstastny) [inline packages/fluentui/docs/src/components/Sidebar/Sidebar.tsx:569]: As we have just one example and 0 results at the moment, I would not add the item to sidebar.

Comment this out, add `//TODO: enable after Nov 2020 when we have data`

File: packages/fluentui/docs/src/routes.tsx [hunk 9]
```diff
@@ -105,13 +107,15 @@ const VirtualizedTablePrototype = React.lazy(() =>
...
 );
+const RosterPrototype = React.lazy(() => import(/* webpackChunkName: "prototypes" */ './prototypes/Roster'));
 
 const Routes = () => (
   <React.Suspense fallback="Loading...">
     {/* Remove trailing slash */}
     <BrowserRouter basename={__BASENAME__ === '/' ? __BASENAME__ : __BASENAME__.slice(0, -1)}>
       <Switch>
         <Route exact path="/maximize/:exampleName/:rtl?" component={ExternalExampleLayout} />
+        <Route exact path="/maximize-compiled/:exampleName/:rtl?" component={PerfExampleLayout} />
```
Comment (miroslavstastny) [inline packages/fluentui/docs/src/routes.tsx:118]: @layershifter recently changed `ExternalExampleLayout` to render compiled examples.
How do `/maximize` and `/maximize-compiled` differ? Can we converge to just one?

Comment (vyhnalekl) [reply packages/fluentui/docs/src/routes.tsx:118]: Changed the code so that the performace popout is using maximize and it seems to be working.

**Review by miroslavstastny — APPROVED** [review]

**Comment by YuanboXue-Amber** [comment]
Close because of #16412

**Pull request closed without merging**